### PR TITLE
Issue #11446: Updated SuppressWarningsHolderTest to use verifyWithInlineConfigParser method

### PR DESCRIPTION
**Pull request by jxr98**
Issue #11446
SuppressWarningsHolder is Annotation check. [documentation](https://checkstyle.org/config_annotation.html#SuppressWarningsHolder)
I don't know how to replace the configs in methods `testAll()`, `testSuppressWarningsTextBlocks()`, `testWithAndWithoutCheckSuffixDifferentCases()` with inlined comment configs. Thanks.

### CODE REVIEW

**Comment by romani** [comment]
@Vyom-Yadav , please help here

**Review by Vyom-Yadav — COMMENTED** [review]
@jxr98 Few points:
1. Do not truncate the title with ..., use the complete commit message in the title.
2. Specify in the PR description, for which module is this PR, preferably specify the link to the documentation too.

Now the concept:
The Inline Config Parser is just a fancy name for specifying the check configuration in the test file itself. You specify the name of checks with their properties in the input file and parser will automatically create checkstyle config for you.

You have to use `// violation .....` comment wherever their is a violation in the input file. See https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralTry.java for eg.

`testAll()` is not working because you have only specified `SuppressWarningsHolder` and no actual filter, you need to use `SuppressWarningsFilter` too. For `testAll()`, it would be:
```java
/*
com.puppycrawl.tools.checkstyle.checks.SuppressWarningsHolder
aliasList = (default)

com.puppycrawl.tools.checkstyle.filters.SuppressWarningsFilter

com.puppycrawl.tools.checkstyle.checks.whitespace.TypecastParenPadCheck
option = space

*/

```

See https://checkstyle.sourceforge.io/config_annotation.html#SuppressWarningsHolder, https://checkstyle.sourceforge.io/config_filters.html#SuppressWarningsFilter, and https://checkstyle.sourceforge.io/config.html#Filters.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/suppresswarningsholder/InputSuppressWarningsHolder6.java [hunk 1]
```diff
@@ -1,3 +1,14 @@
+/*
+com.puppycrawl.tools.checkstyle.checks.SuppressWarningsHolder
+aliasList = (default)
+
+
+com.puppycrawl.tools.checkstyle.checks.whitespace.TypecastParenPad
+option = space
```
Comment (Vyom-Yadav) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/suppresswarningsholder/InputSuppressWarningsHolder6.java:7]: `com.puppycrawl.tools.checkstyle.checks.whitespace.TypecastParenPad` -> `com.puppycrawl.tools.checkstyle.checks.whitespace.TypecastParenPadCheck`

**Comment by jxr98** [comment]
I have passed all tests except `testSuppressWarningsTextBlocks()` in the following code
https://github.com/jxr98/checkstyle/blob/2449b152a2b2987ceaa331135b0b97fa01aeb963/src/test/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolderTest.java#L456-L463
And the corresponding input file is 
https://github.com/jxr98/checkstyle/blob/2449b152a2b2987ceaa331135b0b97fa01aeb963/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/suppresswarningsholder/InputSuppressWarningsHolderTextBlocks.java#L1-L64
The violation line numbers didn't match. I debugged for some time, but still don't know how to make the actualViolationLines match the expectedViolationLines. Any help would be appreciated.

**Comment by romani** [comment]
>The violation line numbers didn't match

As you add comment header, you shift all old lines, you have to update line numbers in "expected" array.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/suppresswarningsholder/InputSuppressWarningsHolder6.java (hunk 1 above)
Comment (Vyom-Yadav) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/suppresswarningsholder/InputSuppressWarningsHolder6.java]: @jxr98 Please don't resolve review items on your own. Reply 'Done', the reviewer will check it and resolve the review item.

**Comment by jxr98** [comment]
> > The violation line numbers didn't match
> 
> As you add comment header, you shift all old lines, you have to update line numbers in "expected" array.

I have tried to update line numbers in the `expected` array, but the `actualViolationLines` didn't change.

**Comment by romani** [comment]
Update and push code, to let me see all together.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/suppresswarningsholder/InputSuppressWarningsHolder6.java (hunk 1 above)
Comment (jxr98) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/suppresswarningsholder/InputSuppressWarningsHolder6.java]: Okay.

**Comment by jxr98** [comment]
> Update and push code, to let me see all together.

Updated. Thanks.

**Comment by romani** [comment]
Please move violation comment to -1 line.
It should be on same line as name of variable

**Comment by jxr98** [comment]
> Please move violation comment to -1 line. It should be on same line as name of variable

Done. Passed the test. But the violation messages will be included in the strings in this case.

**Comment by Vyom-Yadav** [comment]
@jxr98 Please keep the number of commits 1. Squash extra commits into the main one.

**Comment by jxr98** [comment]
> @jxr98 Please keep the number of commits 1. Squash extra commits into the main one.

Okay. Done.

**Review by romani — APPROVED** [review]
Ok to merge

**Pull request merged**